lib/gc/Analysis/MatmulConfigAnalysis.cpp

```diff
@@ -160,6 +156,42 @@ double computationIntensityOnL2Cache(linalg::LinalgOp &linalgOp,
   return 1 / computationIntensity;
 }
 
+// Bufferization may insert more memref.copy/brgemm cannot verify sucessfully
+// and fall back to linalg lower if the buffer is dynamic
+double dynamicBufferizationCost(linalg::LinalgOp &linalgOp,
+                                ArrayRef<uint32_t> shape,
+                                const MatmulConfig &config,
+                                CPUTargetDescriptionAnalysis &sysDesc) {
+  assert(config.MThreads > 0 && "MThreads should be larger than 0");
+  assert(config.NThreads > 0 && "NThreads should be larger than 0");
+  assert(config.innerMostMBlock > 0 &&
+         "innerMostMBlock should be larger than 0");
+  assert(config.innerMostNBlock > 0 &&
+         "innerMostNBlock should be larger than 0");
+  assert(shape.size() >= 3 && "shape.size() should >= 3");
+  uint32_t M = shape[0], N = shape[1];
+  double cost = 0;
+  uint32_t MNumBlock =
+      llvm::divideCeil(M / config.innerMostMBlock, config.MThreads);
+  uint32_t MNumBlockPerThread =
+      llvm::divideCeil(config.MBlock, config.innerMostMBlock);
+  assert(MNumBlockPerThread > 0 && config.MThreads);
```

Comment:
As long as `config.MBlock>0`, this is always true, right?

Reply:
Yes, Use `validateConfig` to replace this part now